### PR TITLE
feat: added image minimum and maximum gc age

### LINKED ISSUE #16317: Expose imageMinimumGCAge and imageMaximumGCAge kubelet config
/kind feature

**1. Describe IN DETAIL the feature/behavior/change you would like to see.**

I would like to configure `imageMinimumGCAge` and `imageMaximumGCAge` in kubelet config with kops
ref: https://kubernetes.io/docs/reference/config-api/kubelet-config.v1beta1/#kubelet-config-k8s-io-v1beta1-KubeletConfiguration

**2. Feel free to provide a design supporting your feature request.**

adding these to the spec the same way as `imageGCHighThresholdPercent` already is

### PR DESCRIPTION
**Pull request by Lerentis**
fixes #16317 

i hope i added these correctly. tried to follow these docs: https://github.com/kubernetes/kops/blob/master/docs/contributing/api_updates.md#adding-a-field

### CODE REVIEW

File: pkg/apis/kops/componentconfig.go [hunk 1]
```diff
@@ -130,6 +130,12 @@ type KubeletConfigSpec struct {
 	// and overrides the default MTU for cases where it cannot be automatically
 	// computed (such as IPSEC).
 	NetworkPluginMTU *int32 `json:"networkPluginMTU,omitempty" flag:"network-plugin-mtu"`
+	// imageMinimumGCAge is the minimum age for an unused image before it is garbage collected. Default: "2m"
+	ImageMinimumGCAge string `json:"imageMinimumGCAge,omitempty" flag:"image-minimum-gc-age"`
```
Comment (hakman) [inline pkg/apis/kops/componentconfig.go:134]:
```suggestion
	ImageMinimumGCAge *string `json:"imageMinimumGCAge,omitempty" flag:"image-minimum-gc-age"`
```

File: pkg/apis/kops/componentconfig.go [hunk 2]
```diff
@@ -130,6 +130,12 @@ type KubeletConfigSpec struct {
...
+	// imageMaximumGCAge is the maximum age an image can be unused before it is garbage collected.
+	// The default of this field is "0s", which disables this field--meaning images won't be garbage
+	// collected based on being unused for too long. Default: "0s" (disabled)
+	ImageMaximumGCAge string `json:"imageMaximumGCAge,omitempty" flag:"image-maximum-gc-age"`
```
Comment (hakman) [inline pkg/apis/kops/componentconfig.go:138]:
```suggestion
	ImageMaximumGCAge *string `json:"imageMaximumGCAge,omitempty" flag:"image-maximum-gc-age"`
```

File: pkg/apis/kops/v1alpha2/componentconfig.go [hunk 3]
```diff
@@ -130,6 +130,12 @@ type KubeletConfigSpec struct {
 	// and overrides the default MTU for cases where it cannot be automatically
 	// computed (such as IPSEC).
 	NetworkPluginMTU *int32 `json:"networkPluginMTU,omitempty" flag:"network-plugin-mtu"`
+	// imageMinimumGCAge is the minimum age for an unused image before it is garbage collected. Default: "2m"
+	ImageMinimumGCAge string `json:"imageMinimumGCAge,omitempty" flag:"image-minimum-gc-age"`
```
Comment (hakman) [inline pkg/apis/kops/v1alpha2/componentconfig.go:134]:
```suggestion
	ImageMinimumGCAge *string `json:"imageMinimumGCAge,omitempty" flag:"image-minimum-gc-age"`
```

File: pkg/apis/kops/v1alpha2/componentconfig.go [hunk 4]
```diff
@@ -130,6 +130,12 @@ type KubeletConfigSpec struct {
...
+	// imageMaximumGCAge is the maximum age an image can be unused before it is garbage collected.
+	// The default of this field is "0s", which disables this field--meaning images won't be garbage
+	// collected based on being unused for too long. Default: "0s" (disabled)
+	ImageMaximumGCAge string `json:"imageMaximumGCAge,omitempty" flag:"image-maximum-gc-age"`
```
Comment (hakman) [inline pkg/apis/kops/v1alpha2/componentconfig.go:138]:
```suggestion
	ImageMaximumGCAge *string `json:"imageMaximumGCAge,omitempty" flag:"image-maximum-gc-age"`
```

File: pkg/apis/kops/v1alpha3/componentconfig.go [hunk 5]
```diff
@@ -129,6 +129,12 @@ type KubeletConfigSpec struct {
 	// and overrides the default MTU for cases where it cannot be automatically
 	// computed (such as IPSEC).
 	NetworkPluginMTU *int32 `json:"networkPluginMTU,omitempty" flag:"network-plugin-mtu"`
+	// imageMinimumGCAge is the minimum age for an unused image before it is garbage collected. Default: "2m"
+	ImageMinimumGCAge string `json:"imageMinimumGCAge,omitempty" flag:"image-minimum-gc-age"`
```
Comment (hakman) [inline pkg/apis/kops/v1alpha3/componentconfig.go:133]:
```suggestion
	ImageMinimumGCAge *string `json:"imageMinimumGCAge,omitempty" flag:"image-minimum-gc-age"`
```

File: pkg/apis/kops/v1alpha3/componentconfig.go [hunk 6]
```diff
@@ -129,6 +129,12 @@ type KubeletConfigSpec struct {
...
+	// imageMaximumGCAge is the maximum age an image can be unused before it is garbage collected.
+	// The default of this field is "0s", which disables this field--meaning images won't be garbage
+	// collected based on being unused for too long. Default: "0s" (disabled)
+	ImageMaximumGCAge string `json:"imageMaximumGCAge,omitempty" flag:"image-maximum-gc-age"`
```
Comment (hakman) [inline pkg/apis/kops/v1alpha3/componentconfig.go:137]:
```suggestion
	ImageMaximumGCAge *string `json:"imageMaximumGCAge,omitempty" flag:"image-maximum-gc-age"`
```

**Review by hakman — CHANGES_REQUESTED** [review]
Thanks for adding this @Lerentis. It looks good mostly.
Please use `*string` instead of `string` and re-run `make apimachinery`.

**Comment by Lerentis** [comment]
> Thanks for adding this @Lerentis. It looks good mostly. Please use `*string` instead of `string` and re-run `make apimachinery`.

Changed and rerun the generators accordingly :smiley:

**Review by hakman — APPROVED** [review]
Thanks @Lerentis! 👍

**Pull request merged**